import pandas as pd
import numpy as np

In [54]:
df= pd.read_csv("/home/lewikeezy/Downloads/Data-for-analysis.csv")
df.head(5)

age  anc1_ordinal anc1_ordinal_date  anc2_ordinal anc2_ordinal_date  \
0   43             1          7/5/2017             1          8/2/2017   
1   32             1         3/22/2017             1          5/7/2017   
2   32             1         12/7/2016             1          1/8/2017   
3   29             1         9/25/2017             0          1/1/1970   
4   26             1          3/2/2017             1          7/4/2017   

   anc3_ordinal anc3_ordinal_date  anc4_ordinal  anc4date  c_section  ...  \
0             1         10/1/2017             0  1/1/1970          0  ...   
1             1          7/5/2017             0  1/1/1970          0  ...   
2             1          2/8/2017             1  3/8/2017          0  ...   
3             0          1/1/1970             0  1/1/1970          1  ...   
4             0          1/1/1970             0  1/1/1970          0  ...   

   previous_perineal_tear previous_placenta_previa previous_pph  \
0                       0                        0            0   
1                       0                        0            0   
2                       0                        0            0   
3                       0                        0            0   
4                       0                        0            0   

  previous_prolonged_labor previous_retained_placenta previous_vacuum  \
0                        0                          0               0   
1                        1                          0               0   
2                        0                          0               0   
3                        0                          0               0   
4                        0                          0               0   

  prior_c_section  sickle_cell stillbirths  twins  
0               0            0           0      0  
1               0            0           0      0  
2               0            0           0      0  
3               0            0           0      0  
4               0            0           0      0  

[5 rows x 61 columns]

In [30]:
df.tail(5)

age  anc1_ordinal anc1_ordinal_date  anc2_ordinal anc2_ordinal_date  \
2995   20             1         8/13/2017             0          1/1/1970   
2996   24             1         8/13/2017             1         9/13/2017   
2997   25             1         1/26/2017             0          1/1/1970   
2998   22             1         6/13/2017             1         7/13/2017   
2999   24             1         9/24/2017             1        11/19/2017   

      anc3_ordinal anc3_ordinal_date  anc4_ordinal    anc4date  c_section  \
2995             0          1/1/1970             0    1/1/1970          0   
2996             1        10/11/2017             1  11/13/2017          0   
2997             0          1/1/1970             0    1/1/1970          0   
2998             0          1/1/1970             0    1/1/1970          0   
2999             0          1/1/1970             0    1/1/1970          0   

      ...  previous_perineal_tear previous_placenta_previa previous_pph  \
2995  ...                       0                        0            0   
2996  ...                       0                        0            0   
2997  ...                       0                        0            0   
2998  ...                       0                        0            0   
2999  ...                       0                        0            0   

     previous_prolonged_labor previous_retained_placenta previous_vacuum  \
2995                        0                          0               0   
2996                        0                          0               0   
2997                        0                          0               0   
2998                        0                          0               0   
2999                        0                          0               0   

     prior_c_section  sickle_cell stillbirths  twins  
2995               0            0           0      0  
2996               0            0           0      0  
2997               0            0           0      0  
2998               0            0           0      0  
2999               0            0           1      0  

[5 rows x 61 columns]

From the above Output of the top 5 rows and bottom 5 rows it is seen that there are 61 colums in total, and this also allows one to have closer onservation of the dataset

In [28]:
df.shape

(3000, 61)

from running df.shape we can see a fair representaion of the total numbers of rows and columns, there are 3000 rows and 61 columns contained in the dataset all together.
This implies that the Dataset comprises of 3000 observations and 61 characteristics
1 is the dependent variable and the remaining 60 are the independent variables

In [31]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3000 entries, 0 to 2999
Data columns (total 61 columns):
 #   Column                      Non-Null Count  Dtype 
---  ------                      --------------  ----- 
 0   age                         3000 non-null   int64 
 1   anc1_ordinal                3000 non-null   int64 
 2   anc1_ordinal_date           3000 non-null   object
 3   anc2_ordinal                3000 non-null   int64 
 4   anc2_ordinal_date           3000 non-null   object
 5   anc3_ordinal                3000 non-null   int64 
 6   anc3_ordinal_date           3000 non-null   object
 7   anc4_ordinal                3000 non-null   int64 
 8   anc4date                    3000 non-null   object
 9   c_section                   3000 non-null   int64 
 10  childdeath                  3000 non-null   int64 
 11  cost_total_facility         3000 non-null   object
 12  dateofdelivery              3000 non-null   object
 13  delivery_facility_name      3000 non-null   obje

It is seen above that the data has only integer and object values
No variable column has missing/null values


In [32]:
df.describe()

age  anc1_ordinal  anc2_ordinal  anc3_ordinal  anc4_ordinal  \
count  3000.000000   3000.000000    3000.00000   3000.000000   3000.000000   
mean     27.380667      0.955667       0.85500      0.571667      0.228000   
std       6.282535      0.205869       0.35216      0.494920      0.419613   
min      10.000000      0.000000       0.00000      0.000000      0.000000   
25%      23.000000      1.000000       1.00000      0.000000      0.000000   
50%      27.000000      1.000000       1.00000      1.000000      0.000000   
75%      31.000000      1.000000       1.00000      1.000000      0.000000   
max      50.000000      1.000000       1.00000      1.000000      1.000000   

         c_section   childdeath  driver_arranged_advance  hivpositive  \
count  3000.000000  3000.000000              3000.000000  3000.000000   
mean      0.032333     0.058333                 0.395000     0.014667   
std       0.176913     0.234412                 0.488932     0.120235   
min       0.000000     0.000000                 0.000000     0.000000   
25%       0.000000     0.000000                 0.000000     0.000000   
50%       0.000000     0.000000                 0.000000     0.000000   
75%       0.000000     0.000000                 1.000000     0.000000   
max       1.000000     1.000000                 1.000000     1.000000   

       hivstatusknown  ...  previous_perineal_tear  previous_placenta_previa  \
count     3000.000000  ...             3000.000000               3000.000000   
mean         0.980333  ...                0.015667                  0.000667   
std          0.138875  ...                0.124203                  0.025816   
min          0.000000  ...                0.000000                  0.000000   
25%          1.000000  ...                0.000000                  0.000000   
50%          1.000000  ...                0.000000                  0.000000   
75%          1.000000  ...                0.000000                  0.000000   
max          1.000000  ...                1.000000                  1.000000   

       previous_pph  previous_prolonged_labor  previous_retained_placenta  \
count   3000.000000               3000.000000                 3000.000000   
mean       0.025333                  0.060333                    0.005333   
std        0.157162                  0.238143                    0.072847   
min        0.000000                  0.000000                    0.000000   
25%        0.000000                  0.000000                    0.000000   
50%        0.000000                  0.000000                    0.000000   
75%        0.000000                  0.000000                    0.000000   
max        1.000000                  1.000000                    1.000000   

       previous_vacuum  prior_c_section  sickle_cell  stillbirths        twins  
count      3000.000000      3000.000000       3000.0  3000.000000  3000.000000  
mean          0.002667         0.021667          0.0     0.021333     0.012000  
std           0.051579         0.145617          0.0     0.171915     0.108903  
min           0.000000         0.000000          0.0     0.000000     0.000000  
25%           0.000000         0.000000          0.0     0.000000     0.000000  
50%           0.000000         0.000000          0.0     0.000000     0.000000  
75%           0.000000         0.000000          0.0     0.000000     0.000000  
max           1.000000         1.000000          0.0     3.000000     1.000000  

[8 rows x 39 columns]

we can observe from the above that for most cases the mean value is greater than the median value (which is represented by the 50th percentile) for must of the observations.
It is obviosu that for the "age" column there is a significant large difference between the 75th percentile and the Max ( a diffrence 19.0000000) and the next most significant difference is seen in the "stillbirths" column (a difference of 2.000000).
This indicates that there are exisits some extreme values outliers in the data

In [47]:
df.stillbirths.unique()

array([0, 1, 3, 2])

 From the output above the variable "stillbirths" is discrete and categorical in nature.
 The  "stillbirths score scale ranges from 0 to 3 ; where 3 being poor and 0 being the best.
 where 3 shows the highest number of stillbirths in the dataset


In [49]:
df.stillbirths.value_counts()

0    2948
1      41
2      10
3       1
Name: stillbirths, dtype: int64

The above output shows us the  vote count of each stillbirths score in descending order.
we can see that 2948 people did not have still births
41 persons experienced a single still birth, 10 people had at least 2 still births while 1 person only had 3 still births


Recommendations
1. Looking at the observation from the still births column,what can be done to reduce the still births to a barest minimal
2. Identify the reasons or factors why some people do not book or make preparations for drivers or motors in advnce for their delivery date
3 Are there prevailing health conditions that xan jeopadise the safe deliverly process? if so what can be done address the issue
